pennylane/tape/tape.py

```diff
@@ -1036,12 +1037,70 @@ def get_depth(self):
 
         >>> tape.get_depth()
         4
+
         """
+
+        warnings.warn(
+            "``tape.get_depth`` will be deprecated after v0.16 "
+            "Please use the more general ``tape.specs`` instead.",
+            UserWarning,
+        )
+
         if self._depth is None:
             self._depth = self.graph.get_depth()
 
         return self._depth
 
+    @property
+    def specs(self):
+        """Resource requirements of a quantum circuit.
```

Comment:
These seem to be general info rather than requirements, right? Otherwise not sure what a resource requirement of a quantum circuit would be (maybe the resource requirement of executing a quantum circuit?).

Or using the word "specification" could also be useful as coming from the name.
```suggestion
        """Resource information about a quantum circuit.
```